recipe/patches/0010-Allow-overriding-CUDA-related-paths.patch

```diff
@@ -0,0 +1,39 @@
+From 40f56d52a14d72068f62c685a5fdf26727150a6a Mon Sep 17 00:00:00 2001
+From: =?UTF-8?q?Micha=C5=82=20G=C3=B3rny?= <mgorny@gentoo.org>
+Date: Wed, 27 Nov 2024 13:47:23 +0100
+Subject: [PATCH] Allow overriding CUDA-related paths
+
+---
+ cmake/Modules/FindCUDAToolkit.cmake | 2 +-
+ tools/setup_helpers/cmake.py        | 2 +-
+ 2 files changed, 2 insertions(+), 2 deletions(-)
+
+diff --git a/cmake/Modules/FindCUDAToolkit.cmake b/cmake/Modules/FindCUDAToolkit.cmake
+index ec9ae530..76d1fb9a 100644
+--- a/cmake/Modules/FindCUDAToolkit.cmake
++++ b/cmake/Modules/FindCUDAToolkit.cmake
+@@ -497,7 +497,7 @@ Result variables
+ # CMAKE_CUDA_COMPILER_TOOLKIT_ROOT and CMAKE_CUDA_COMPILER_LIBRARY_ROOT.
+ # We compute the rest based on those here to avoid re-searching and to avoid finding a possibly
+ # different installation.
+-if(CMAKE_CUDA_COMPILER_TOOLKIT_ROOT)
++if(FALSE)
+   set(CUDAToolkit_ROOT_DIR "${CMAKE_CUDA_COMPILER_TOOLKIT_ROOT}")
```

Comment:
is there a way to make this patch "upstreamable"? While they have been slow to merge patches they do work to merge them eventually....

Reply:
That's a very good question. I suppose the cleanest approach would be to add a new variable to control/override this behavior — though I would find it hard to justify it beyond "this is the simplest way I managed to make the current deeply complex logic let me use a separate 'host' tree for conda".

Reply:
It’s funny because I literally wrote the same message elsewhere. 

Maybe we can open an internal tracking issue to discuss longer term solutions. Upstream has historically periodically compared their build systems to ours. But it is unclear how this is evolving as they dropped conda support.

Reply:
I think the biggest problem here is that we're seeing an intersection of three different sets of logic:

1. Built-in CMake support for using the CUDA compiler.
2. The `FindCUDA` module with its patched version inside PyTorch.
3. The `FindCudaToolkit` module within PyTorch.

I think what upstream is really trying to do here is gluing they all together, so they don't end up using different CUDA installations, but I don't know how many different use cases they are trying to cover here. On top of that, I think Conda is a bit of an "odd one" here, using separate build and host trees even when not actually cross-compiling.